src/main/java/org/opentripplanner/routing/edgetype/TemporaryEdge.java

```diff
@@ -1,6 +1,4 @@
 package org.opentripplanner.routing.edgetype;
 
 /** Marker interface for temporary edges */
-public interface TemporaryEdge {
-    public void dispose();
-}
+public interface TemporaryEdge { }
```

Comment:
Having an interface without any methods seems like an anti-pattern. Is there another strategy that could be used here? For example, instead, Vertex could have a method `isTemporary()` which has a default implementation of false; subclasses which are temporary would override it.

Reply:
I agree @sdjacobs. I do not like marker interfaces either, it is better to add methods to the Vetex and Edge, and avoid the `instanceof`. I would be happy to clean it up. But I would like to NOT do it as part of this relatively small bug fix.

Reply:
I see @t2gran has created issue #2695 to cover this. I agree that it can be done later as a style fix. This PR has been tested as it currently stands, and any such refactors would require repeating the testing.